src/util.rs

```diff
@@ -54,5 +82,52 @@ impl<'r> RepositoryUtil<'r> for Repository {
         }
         Ok(retval)
     }
+
+    fn commitmsg_edit_path(&self) -> PathBuf {
+        self.path().with_file_name("COMMIT_EDITMSG")
+    }
+
+    fn get_commit_msg(&self, path: PathBuf) -> Result<Vec<String>> {
+        // let the user write the message
+        if !run_editor(self.config().chain_err(|| EK::WrappedGitError)?, &path)?
+            .wait().chain_err(|| EK::WrappedIOError)?
```

Comment:
The `chain_err()` belongs to the `wait()`. Consider a chain such as:
```
.foo().chain_err(...)?
.bar().chain_err(...)?
```

Writing it as
```
.foo()
.chain_err(...)?
.bar()
.chain_err(...)?
```
may confuse readers.